fml/memory/thread_checker.h

```diff
@@ -67,7 +78,10 @@ class ThreadChecker final {
     return is_creation_thread_current;
   }
 
+  static void DisableNextThreadCheckFailure() { disable_next_ = true; }
```

Comment:
This isn't safe since we can't deterministically know what thread check is next, so we can't reasonably disable it.

Can you make `disable_next_` thread local?

Reply:
This is only meant to be used in tests so that we can intentionally violate a thread check.

Making it thread local sounds good.